frappe/utils/data.py

```diff
@@ -419,7 +419,10 @@ def money_in_words(number, main_currency = None, fraction_currency=None):
 	from frappe.utils import get_defaults
 	_ = frappe._
 
-	if not number or flt(number) < 0:
+	if not isinstance(number, float) and not isinstance(number, int):
```

Comment:
General philosophy is to be forgiving. If the number is in string, `flt` will convert it to float.